tools/hooks/lint.rb

```diff
@@ -8,16 +8,10 @@
 SCSS_GLOB = "#{REPO_DIR}/#{YAML.load_file('.scss-lint.yml')['scss_files'] || '*'}".freeze
 
 def filter_eslint_apps(modified_files)
+  full_apps_dir = File.expand_path(APPS_DIR)
+
   modified_files.select do |f|
-    (f.end_with?(".js", ".jsx", ".ts", ".tsx")) &&
-      !(f.end_with?('.min.js') ||
-        f.match(/public\/.+package\//) ||
-        f.include?('apps/lib/') ||
-        f.include?('shared/') ||
-        f.include?('dashboard/config/') ||
-        f.include?('pegasus/sites/virtual') ||
-        f.include?('dashboard/public')
-      )
+    f.start_with?(full_apps_dir) && f.end_with?(".js", ".jsx", ".ts", ".tsx")
   end
 end
```

Comment:
Unlike the other filter_* methods in lint.rb, `filter_eslint_apps()` is missing the most basic directory check, which causes it to enforce /apps eslint rules across the repo, with a list of exceptions added one by one (lines 14 - 19), but still actually doesn't cover all files it mis-includes, like pegasus/sites.v3.

Compare with the code for other `filter_*` methods like this:
```
def filter_scss_apps(modified_files)
  modified_files.select {|f| f.include?('apps/') && f.end_with?(".scss")}
end
```

Or:
```
def filter_eslint_shared(modified_files)
  modified_files.select do |f|
    f.match(%r{/shared/js/[^/]+.js})
  end
end
```

I think this is a very old bug that we never fixed.

As a result of this bug, there are a lot of JS files outside /apps that if they were ever modified would be failed by this pre-commit hook (many of them are pre-bundled libraries, and it def shouldn't apply). Some of the files that would be mis-failed by this hook due to its missing directory check are:
```
/aws/cloudformation/ami-manager.js
/aws/cloudformation/cfn-response.js
/aws/cloudformation/count_asg.js
/aws/cloudformation/honeybadgerNotify.js
/aws/cloudformation/lambdas/sql-user/index.js
/aws/cloudformation/lambdas/sql-user/index.test.js
/aws/cloudformation/s3Object.js
/aws/cloudformation/s3PartitionCloudFrontLog.js
/aws/cloudformation/slackCloudWatchEvent.js
/aws/cloudformation/test/index.js
/aws/cloudformation/test/testHoneybadgerNotify.js
/aws/cloudformation/test/testS3PartitionCloudFrontLog.js
/aws/cloudformation/vpcClassicLink.js
/aws/offsite/aurora-snapshots-fargate/restore-and-verify-backup.js
/aws/offsite/aurora-snapshots-fargate/test/test-restore-and-verify-backup.js
/aws/offsite/prune-aurora-backups/src/prune-aurora-backups.js
/aws/offsite/prune-aurora-backups/src/tests/test-prune-aurora-backups.js
/bin/generate-pdf/generatePdf.js
/bin/oneoff/pullOutBlocks.js
/dashboard/scripts/archive/migrateAnimationData.js
/dashboard/scripts/archive/zendesk_theme_js.js
/dashboard/test/ui/js/set-tab-status-icon.js
/dashboard/vendor/assets/javascripts/add2home.js
/lib/cdo/pegasus/screencap.js
/lib/pdf/pdf_rasterize.js
/tools/scripts/brokenLinkChecker/brokenLinkChecker.js
/tools/scripts/prune-rds-backups.js
```

Reply:
Do the files that are currently included in this eslint file but will be ignored on future lint runs need their own eslint config or some way to get linted similarly to apps?